### ISSUE TITLE
change event not firing on email input in React 16

### ISSUE THREAD

**Issue by chodorowicz** [cypress-io/cypress#816]
- Operating System: MacOS
- Cypress Version: `1.0.2`
- Browser Version: Electron / Chrome 61

### Is this a Feature or Bug?
Bug

### Current behavior:
When running test via headless Electron change event is not fired for `email` type input. For `text` input it works correctly. Also `text` and `input` work correctly in standard Chrome.

### Desired behavior:
Cypress should fire change event.

### How to reproduce:
Run test `cypress run` on following app: https://stackblitz.com/edit/react-eadssh?file=index.js (need to export it and run locally, embedding doesn't seem to work)

```jsx
import React, { Component } from 'react';
import { render } from 'react-dom';

class App extends Component {
  state = {
    inputValue: '',
    message: ''
  };

  handleChange = (event) => {
    this.setState({ 
      inputValue: event.currentTarget.value,
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
      message: this.state.inputValue === ''
        ? 'Error'
        : 'All good'
    });
  }

  render() {
    return (
      <div>
        input value: {this.state.inputValue}
        <form onSubmit={this.handleSubmit}>
          <input onChange={this.handleChange} value={this.state.inputValue} id="input" type="email" />
          <button>submit</button>
        </form>
        {this.state.message}
      </div>
    );
  }
}

render(<App />, document.getElementById('root'));
```

#### Test code:

```js
describe('app', () => {
  it('should work with input email type', () => {
    cy.visit('http://localhost:3000/');
    cy.get('#input').type('user@example.com');
    cy.contains('submit').click();
    cy.contains('All good');
  });
});
```

### Additional Info (images, stack traces, etc)
related tickets:
- https://github.com/cypress-io/cypress/issues/647
- https://github.com/cypress-io/cypress/issues/770
- test written only for standard input type https://github.com/cypress-io/cypress/commit/4a56ca83b3b6e19ec57d10a8160f54f513e8f8ec

failing with input type `email`
![input-email](https://user-images.githubusercontent.com/119451/32045401-6b19a07e-ba40-11e7-8e0f-bb245903b693.gif)

all good with `text` input type
![input-text](https://user-images.githubusercontent.com/119451/32045423-77555ef0-ba40-11e7-8b12-b43aae24e69c.gif)

**Comment by erkarl** [comment]
Downgrading to `1.0.1` solved the issue for me.

**Comment by jennifer-shehane** [comment]
Related to #770

**Comment by chodorowicz** [comment]
For me with `1.0.1` it doesn't work at all, even in GUI Chrome 61. `1.0.2` was supposed to fix issues with `type` not firing change events, so it makes sense it doesn't work for me at all.

**Comment by brian-mann** [comment]
I'll take a look at this / fix it and we'll release `1.0.3`.

We added a bunch of react tests in `1.0.2` (you all can look at the code) and we run our driver tests in both Electron and Chrome, so I'm curious how this could have gotten missed.

Appreciate you all creating a reproducible example. 

The new tests for react 16 are here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/test/cypress/integration/e2e/react-16_spec.coffee

**Comment by jennifer-shehane** [comment]
We are also experiencing this issue within our own Dashboard Service project which uses React 15 on Cypress version 1.0.2.

Failing, reproducible spec here: https://github.com/cypress-io/cypress-dashboard/blob/develop/cypress/integration/users_spec.coffee#L266

**Comment by chodorowicz** [comment]
@brian-mann Thanks for the answer! What would be a good *reproducible example*? I've pasted code and test samples? Should I fork the repo and add failing test case?

**Comment by Fjoggs** [comment]
I left a comment on this issue on the already closed issue that supposedly solved it, but didn't receive a response. I'm not too familiar with github issues, so I guess I should have just opened a new one? :)

Anyway, I have the same problem, using React 16 and Cypress 1.0.2.
Event's aren't firing for input type="number" using Electron.

**Comment by brian-mann** [comment]
Fixed in [`1.0.3`](https://docs.cypress.io/guides/references/changelog.html#1-0-3).

**Comment by chodorowicz** [comment]
Great, thank you 🙇
